airflow-core/src/airflow/api_fastapi/common/exceptions.py

```diff
@@ -61,13 +61,10 @@ def __init__(self):
     def exception_handler(self, request: Request, exc: IntegrityError):
         """Handle IntegrityError exception."""
         if self._is_dialect_matched(exc):
+            error_message = self._get_user_friendly_message(exc)
             raise HTTPException(
                 status_code=status.HTTP_409_CONFLICT,
-                detail={
```

Comment:
I don't think we should change this with a string. Some tests and logic depend on this is expected to be a dictionary, not a string, in the API and API tests. Is there a reason for this change?